### PR TITLE
feat: ✨ Mint button in confirm modal for drops

### LINKED ISSUE #10150: Mint button in confirm modal for drops
from what i remember, we had this cute button animation for free drops 
image
image

can we please use it for 
image

(main button in confirmation modal when minting) 

IMO @hassnian was doing the animation before. Could you please have a look 👀

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #10150

## Needs QA check

- @kodadot/qa-guild please review

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI;

### CODE REVIEW

**Review by exezbcz — APPROVED** [review]

**Comment by exezbcz** [comment]
woho nice

File: libs/ui/src/components/NeoButton/NeoButton.scss [hunk 1]
```diff
@@ -47,6 +48,14 @@
       &.o-btn--disabled {
         @apply opacity-unset bg-k-shade border-k-grey text-k-grey;
       }
+
+      &.shiny:not(:hover):not(:disabled) {
+        @include shineEffect(var(--k-accent-light-3), lightgrey, false);
+
+        &:hover {
+            color: var(--k-accent2) !important;
```
Comment (Jarsen136) [inline libs/ui/src/components/NeoButton/NeoButton.scss:56]: Could you try with it? `@apply hover:text-k-accent2`

**Review by Jarsen136 — APPROVED** [review]
Nice! It's easy to reuse this animation effect on other buttons now.

**Comment by prury** [comment]
[Gravação de tela de 26-04-2024 14:56:42.webm](https://github.com/kodadot/nft-gallery/assets/36627808/1b2a05fa-1500-46ff-afa0-1236b3bd7653)

@exezbcz do you think its too fast on mobile or nah? 
desktop looks nice

File: libs/ui/src/components/NeoButton/NeoButton.scss (hunk 1 above)
Comment (hassnian) [reply libs/ui/src/components/NeoButton/NeoButton.scss]: updated

**Review by preschian — APPROVED** [review]

**Pull request merged**